src/screen.rs

```diff
@@ -1114,22 +1118,53 @@ impl Screen {
             Cancel => {
                 self.error_file = None;
                 self.set_search(None);
+                self.clear_repeat_count();
                 self.error = None;
                 self.refresh();
                 return DisplayAction::ClearOverlay;
             }
             PreviousFile => return DisplayAction::PreviousFile,
             NextFile => return DisplayAction::NextFile,
-            ScrollUpLines(n) => self.scroll_up(n),
-            ScrollDownLines(n) => self.scroll_down(n),
-            ScrollUpScreenFraction(n) => self.scroll_up_screen_fraction(n),
-            ScrollDownScreenFraction(n) => self.scroll_down_screen_fraction(n),
-            ScrollToTop => self.scroll_to(0),
-            ScrollToBottom => self.following_end = true,
-            ScrollLeftColumns(n) => self.scroll_left(n),
-            ScrollRightColumns(n) => self.scroll_right(n),
-            ScrollLeftScreenFraction(n) => self.scroll_left_screen_fraction(n),
-            ScrollRightScreenFraction(n) => self.scroll_right_screen_fraction(n),
+            ScrollUpLines(n) => {
+                let n = self.apply_repeat_count(n);
+                self.scroll_up(n)
+            }
+            ScrollDownLines(n) => {
+                let n = self.apply_repeat_count(n);
+                self.scroll_down(n)
+            }
+            ScrollUpScreenFraction(n) => {
+                let n = self.apply_repeat_count(n);
+                self.scroll_up_screen_fraction(n)
+            }
+            ScrollDownScreenFraction(n) => {
+                let n = self.apply_repeat_count(n);
+                self.scroll_down_screen_fraction(n)
+            }
+            ScrollToTop => self.scroll_to(self.repeat_count.unwrap_or(0)),
+            ScrollToBottom => {
+                if let Some(n) = self.repeat_count {
+                    self.scroll_to(n);
```

Comment:
This seems weird, but it's what less does, so OK.

Reply:
This is more about vim - G: bottom; 10G: go to line 10; 10g: incomplete vim command.